### PR TITLE
[TT-10221] Update Config structure

### PR DESCRIPTION
**Pull request by buraksekili**
## Description
This PR adds `envconfig` to expose consistent environment variables for Tyk K8s Bootstrap. Also, it updates bootstrap configuration structure. 

### Removed fields
- `DashboardHost`,
- `CatalogId`,
- `OperatorSecretEnabled`
- `DeveloperPortalSecretEnabled`

### Renamed fields
- Cname => TykOrgCname,
- TYK_POD_NAMESPACE => ReleaseNamespace
- DashboardSvc => DashboardSvcName
- DashboardInsecureSkipVerify InsecureSkipVerify
- IsDashboardEnabled => DashboardEnabled

## Related Issue

https://tyktech.atlassian.net/browse/TT-10221

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] Make sure you are updating [CHANGELOG.md](../CHANGELOG.md) based on your changes.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `gofmt -s -w .`
  - [ ] `go vet ./...`
  - [ ] `golangci-lint run`

### CODE REVIEW

File: data/config.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	// InsecureSkipVerify enables InsecureSkipVerify options in http request sent to Tyk - might be useful
+	// for Tyk Dashboard with self-signed certs.
+	InsecureSkipVerify bool
+	// BootstrapDashboard controls bootstrapping Tyk Dashboard or not.
+	BootstrapDashboard bool
+	// BootstrapPortal controls bootstrapping Tyk Classic Portal or not.
+	BootstrapPortal bool
+
+	// OperatorKubernetesSecretName corresponds to the Kubernetes secret name that will be created for Tyk Operator.
+	// Set it to an empty to string to disable bootstrapping Kubernetes secret for Tyk Operator.
```
Comment (komalsukhani) [inline data/config.go]:
```suggestion
	// Set it to an empty string to disable bootstrapping Kubernetes secret for Tyk Operator.
```

File: data/config.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	// By default, tyk-operator-conf
+	OperatorKubernetesSecretName string
+	// DevPortalKubernetesSecretName corresponds to the Kubernetes secret name that will be created for
+	// Tyk Developer Enterprise Portal. Set it to an empty to string to disable bootstrapping Kubernetes
```
Comment (komalsukhani) [inline data/config.go]:
```suggestion
	// Tyk Developer Portal. Set it to an empty to string to disable bootstrapping Kubernetes
```

File: data/config.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	// secret for Tyk Developer Enterprise Portal.
```
Comment (komalsukhani) [inline data/config.go]:
```suggestion
	// secret for Tyk Developer Portal.
```

File: README.md [hunk 4]
```diff
@@ -1,39 +1,61 @@
...
-This is a standalone app meant to help with bootstrap and deletion of the tyk-stack when installed
-via tyk-helm-charts.
+This is a standalone app meant to help with bootstrap and deletion of the [tyk-stack](https://github.com/TykTechnologies/tyk-charts/tree/main/tyk-stack) when installed
+via Helm Charts
 
-**Note:** 
-<br>This app is needed only for Tyk [Self-managed](https://tyk.io/docs/tyk-on-premises/) deployment!
-<br>[Tyk OSS](https://tyk.io/docs/apim/open-source/) doesn't have a special bootstrap and in [Tyk Cloud](https://tyk.io/docs/tyk-cloud/) it is done for you (being a SaaS).
+> [!NOTE]
+This app is needed only for [Tyk Self-Managed](https://tyk.io/docs/tyk-on-premises/) deployment!
```
Comment (komalsukhani) [inline README.md]: It's not true anymore. Since we can use bootstrap app to create operator secret.

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: data/config.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+	// for Tyk Dashboard with self-signed certs.
+	InsecureSkipVerify bool
+	// BootstrapDashboard controls bootstrapping Tyk Dashboard or not.
+	BootstrapDashboard bool
+	// BootstrapPortal controls bootstrapping Tyk Classic Portal or not.
+	BootstrapPortal bool
+
+	// OperatorKubernetesSecretName corresponds to the Kubernetes secret name that will be created for Tyk Operator.
+	// Set it to an empty to string to disable bootstrapping Kubernetes secret for Tyk Operator.
+	// By default, tyk-operator-conf
```
Comment (komalsukhani) [inline data/config.go]: If I have understood correctly, these default values set in the helm chart.
If so, we should remove these default value comment.

Comment (buraksekili) [reply data/config.go]: updated 👍

File: README.md (hunk 4 above)
Comment (buraksekili) [reply README.md]: updated 👍

**Review by komalsukhani — APPROVED** [review]

**Comment by singhpr** [comment]
tested this up, it works as expected , hence merging this up.

**Pull request merged**